### ISSUE TITLE
`decompose_essential_matrix` / `motion_from_essential` batch the rotations of a `(3, 3)` input but not `t`

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4878]
`decompose_essential_matrix(E)` documents its outputs as `[(*, 3, 3), (*, 3, 3), (*, 3, 1)]`, "same input". For an unbatched `(3, 3)` input it returns rotations of shape `(1, 3, 3)` and `t` of shape `(3, 1)`: the constant `W` is built batched, `cross_product_matrix(torch.tensor([[0.0, 0.0, 1.0]]))` of shape `(1, 3, 3)` (`essential.py:528`), and `U @ W @ Vt` broadcasts the unbatched `U` against it. `motion_from_essential` inherits the mismatch and returns `Rs` of shape `(1, 4, 3, 3)` next to `ts` of shape `(4, 3, 1)`. Batched inputs `(1, 3, 3)` and `(2, 1, 3, 3)` keep their leading dims. `motion_from_essential_choose_solution` adds its own batch dim before calling and is unaffected. Expected, per the docstrings: `(3, 3)`, `(3, 3)`, `(3, 1)` and `(4, 3, 3)`, `(4, 3, 1)`.

## Repro

```python
import torch
from kornia.geometry.conversions import axis_angle_to_rotation_matrix
from kornia.geometry.epipolar import decompose_essential_matrix, essential_from_Rt, motion_from_essential

R = axis_angle_to_rotation_matrix(torch.tensor([[0.1, -0.2, 0.05]]))
t = torch.tensor([[[0.5], [0.1], [0.02]]])
E = essential_from_Rt(torch.eye(3)[None], torch.zeros(1, 3, 1), R, t)   # (1, 3, 3)

for name, e in [("(3, 3)", E[0]), ("(1, 3, 3)", E), ("(2, 1, 3, 3)", E.expand(2, 1, 3, 3))]:
    R1, R2, tt = decompose_essential_matrix(e)
    Rs, ts = motion_from_essential(e)
    print(f"{name:13s} decompose -> R1 {tuple(R1.shape)}, R2 {tuple(R2.shape)}, t {tuple(tt.shape)} | "
          f"motion -> Rs {tuple(Rs.shape)}, ts {tuple(ts.shape)}")
```

Output:

```text
(3, 3)        decompose -> R1 (1, 3, 3), R2 (1, 3, 3), t (3, 1) | motion -> Rs (1, 4, 3, 3), ts (4, 3, 1)
(1, 3, 3)     decompose -> R1 (1, 3, 3), R2 (1, 3, 3), t (1, 3, 1) | motion -> Rs (1, 4, 3, 3), ts (1, 4, 3, 1)
(2, 1, 3, 3)  decompose -> R1 (2, 1, 3, 3), R2 (2, 1, 3, 3), t (2, 1, 3, 1) | motion -> Rs (2, 1, 4, 3, 3), ts (2, 1, 4, 3, 1)
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same), CPU float32, macOS arm64.

## Classification

Clearly broken output (the returned tensors disagree in their leading dims). No idiom of 1–9; candidate: a constant built with a batch dim broadcasts an unbatched input.

## Disposition

MERGE IN WINDOW — behaviour change: build `W` unbatched, `[[0, -1, 0], [1, 0, 0], [0, 0, 1]]` of shape `(3, 3)` in `E_mat`'s dtype and device, so a `(3, 3)` input returns `(3, 3)` rotations and `motion_from_essential` returns `(4, 3, 3)`, `(4, 3, 1)`. A caller that indexes `R1[0]` after an unbatched call has to drop the index.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
